homeassistant/components/splunk/__init__.py

```diff
@@ -55,45 +55,47 @@
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the Splunk component."""
-    conf = config[DOMAIN]
-    host = conf.get(CONF_HOST)
-    port = conf.get(CONF_PORT)
-    token = conf.get(CONF_TOKEN)
-    use_ssl = conf[CONF_SSL]
-    verify_ssl = conf.get(CONF_VERIFY_SSL)
-    name = conf.get(CONF_NAME)
-    entity_filter = conf[CONF_FILTER]
-
-    event_collector = hass_splunk(
+    """Old setup for Splunk component."""
+    if DOMAIN in config:
+        # Entity filters are not configurable in Config Flow, so are removed
+        del config[DOMAIN]["filter"]
+        _LOGGER.warning(
+            "Your Splunk configuration has been imported into the UI; "
+            "please remove it from configuration.yaml as support for it "
+            "will be removed in a future release"
+        )
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Splunk from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    host = entry.data[CONF_HOST]
+    port = entry.data[CONF_PORT]
+    name = entry.data.get(CONF_NAME)
+    splunk = hass_splunk(
         session=async_get_clientsession(hass),
         host=host,
         port=port,
-        token=token,
-        use_ssl=use_ssl,
-        verify_ssl=verify_ssl,
+        token=entry.data[CONF_TOKEN],
+        use_ssl=entry.data[CONF_SSL],
+        verify_ssl=entry.data[CONF_VERIFY_SSL],
     )
 
-    if not await event_collector.check(connectivity=False, token=True, busy=False):
-        return False
-
-    payload = {
-        "time": time.time(),
-        "host": name,
-        "event": {
-            "domain": DOMAIN,
-            "meta": "Splunk integration has started",
-        },
-    }
-
-    await event_collector.queue(json.dumps(payload, cls=JSONEncoder), send=False)
+    if not await splunk.check(connectivity=False, token=True, busy=False):
```

Comment:
If we can't make a connection, how do we know that the token is ok?

Reply:
We dont. However a lack of connectivity is most likely a recoverable problem, for example Splunk is still starting up (if the server was rebooted).

I can fail setup if thats required, but it does mean there will be some data lost if Home Assistant starts up faster than Splunk.

Reply:
What do we want to do here? Currently I don't understand the parameters since they are not able to be combined like this.

Reply:
The way its written now is the same as the YAML configuration. If there is no connectivity it gets setup, collects data, and keeps retrying. Same is true if busy.

If the token is invalid setup the YAML failed, but in Config Flow we trigger reauth.

Reply:
For reference, the YAML setup was:
```
if not await event_collector.check(connectivity=False, token=True, busy=False):
        return False

Reply:
I think we should it like this.

1. Try to connect, if connection fails raise `ConfigEntryNotReady`.
2. Try to authenticate, if authentication fails raise `ConfigEntryFauthFailed`.

Reply:
Thinking a bit more, I think it's ok to keep it like it is, since we raise `ConfigEntryNotReady` later also. But add a comment explaining that it's ok to not have a connection during config entry setup, since there's a queue for the events and we'll keep trying sending the events from the queue.